luminex/src/org/labkey/luminex/LuminexExcelParser.java

```diff
@@ -242,6 +253,93 @@ private void parseFile() throws ExperimentException
         _parsed = true;
     }
 
+    // Keeps track of potential titrations using the following structure. 'count_descriptionM_analyteN' indicates the number of instances where descriptionM and analyteN are found on the same row.
+    // crossFilePT: {description1: {analyte1: count_description1_analyte1, analtye2: count_description1_analyte2, ...},
+    //               description2: {analyte1: count_description2_analyte1, analyte2: count_description2_analyte2, ...} ... }
+    private Map<String, HashMap<String, Integer>> buildCrossFilePTs(Map<String, HashMap<String, Integer>> crossFilePT, Map<String, Titration> potentialTitrations, String analyteName, Map<String, Integer> potentialTitrationCounts, String type)
+    {
+        for (String desc : potentialTitrations.keySet())
+        {
+            if (crossFilePT.containsKey(desc))
+            {
+                if (crossFilePT.get(desc).containsKey(analyteName))
+                {
+                    Integer count = crossFilePT.get(desc).get(analyteName);
+                    crossFilePT.get(desc).put(analyteName, count + potentialTitrationCounts.get(desc));
+                }
+                else
+                {
+                    crossFilePT.get(desc).put(analyteName, potentialTitrationCounts.get(desc));
+                }

```

Comment:
If we do the refactor suggested above about removing the analyte name part of the map, this can probably be simplified to just make sure the given desc is in the crossFilePT map and then add on the counts for that desc.

Reply:
Done